client/network/src/service.rs

```diff
@@ -239,11 +225,8 @@ where
 			params.protocol_id.clone(),
 			&params.fork_id,
 			&params.network_config,
-			iter::once(Vec::new())
-				.chain(
-					(0..params.network_config.extra_sets.len() - 1)
-						.map(|_| default_notif_handshake_message.clone()),
-				)
+			(0..params.network_config.extra_sets.len())
+				.map(|_| default_notif_handshake_message.clone())
```

Comment:
Did you test if this is not a breaking change? What happens if the node now starts sending a handshake for the tx protocol?

Reply:
TX protocol is now removed from the extra sets so it doesn't send a handshake anymore.

Reply:
I think it has to be part of it because otherwise we don't advertise the support for the TX protocol. CI also got upset after I removed the TX protocol from `extra_sets`.

Reply:
You still add the tx protocol to `extra_sets`? Just now in a different place. This means that tx protocol will now use the default notif handshake?

Reply:
Resolving, I tested and it worked with the default handshake as well.